builtin/tfc/config_sourcer.go

```diff
@@ -333,10 +344,57 @@ func (cs *ConfigSourcer) read(
 			cs.secretCache[id] = cachedSecretVal
 		}
 
-		value := cachedSecretVal.Outputs[tfcReq.Output]
+		if tfcReq.Output == "" {
+			// No explicit output specified - getting all outputs.
+
+			// Convert the decorated tfc api response types to a native map[string]interface{}
+			allOutputsMap := make(map[string]interface{})
+			for k, v := range cachedSecretVal.Outputs {
+				allOutputsMap[k] = v.Value
+			}
+
+			allOutputsJson, err := json.Marshal(allOutputsMap)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed marshalling all TFC outputs for org %q workspace %q to json", tfcReq.Organization, tfcReq.Workspace)
+			}
+
+			result.Result = &pb.ConfigSource_Value_Json{
+				Json: allOutputsJson,
+			}
+		} else {
+			value := cachedSecretVal.Outputs[tfcReq.Output]
+
+			switch value.Type {
+			case "string":
+				stringVal, ok := value.Value.(string)
+				if !ok {
+					return nil, fmt.Errorf("variable %s is a string according to TFC, but is not actually a string type", value.Name)
```

Comment:
```suggestion
					return nil, fmt.Errorf("variable %q is a string according to TFC, but is not actually a string type", value.Name)
```

Reply:
Maybe also include the type? `value.Type`

Reply:
Done!